src/note/note.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable, Logger } from '@nestjs/common';
+import {InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { notes } from './note.entity';
+import { Note } from '../interfaces/note'
+@Injectable()
+export class  NoteService {
+constructor(
+@InjectRepository(notes)
+private noteRepository: Repository<notes>,
+) {}
+async index(): Promise<Note[]> {
+return await this.noteRepository.find();
+}
+async findOne(id): Promise<Note[]>{
+return await this.noteRepository.find({where: [{ 'id': id }]});
+}
+async create(note: Note): Promise<any> {
+ const noteReady = await this.noteRepository.save(note);
+// Logger.log(noteReady);
+}
+async updata(id: string, body: Note): Promise<any>{
```

Comment:
literówka do poprawy updat**a** na updat**e**

id: string do poprawy na id: number

typ zwracany do poprawy na `Promise<Note>`

Reply:
Poprawiłem ale jeśli chodzi o zwracany typ mam wątpliwości czy dobrze. Update zwraca  obiekt odpowiedzi: {
  "generatedMaps": [],
  "raw": {
    "fieldCount": 0,
    "affectedRows": 0,
    "insertId": 0,
    "serverStatus": 2,
    "warningCount": 0,
    "message": "(Rows matched: 0  Changed: 0  Warnings: 0",
    "protocol41": true,
    "changedRows": 0
  },
  "affected": 0
}
Nawet jeżeli próbuje aktualizować o nieistniejącym id,   nie powinien chyba zwracać  kod 200 ...? Więc zrobiłem zwrotkę z body ale chyba to nie o to chodzi ..?
Jeśli chodzi o zwrotki z endpointów które zmieniają stan BD urodziło mi się jakiś czas temu pytanie czy powinny zwracać coś  więcej jeżeli sukces ?  powiodło się i tyle a resztę może sobie dorobić chyba front end ?

Reply:
Czy odpowiedź na to jest standardowa "to zależy..."  ;)

Reply:
Hm... jeżeli usunąć async i await jak pisałeś powyżej to metoda nie chce mi zwrócić obiektu o typie Promise<Note> Ostatecznie Pozostałem przy wersji Promise<any>

Reply:
Ale takiego obiektu w odpowiedzi chyba te nie powinniśmy wysyłać ????